src/meta-srv/src/procedure/region_migration/downgrade_leader_region.rs

```diff
@@ -55,11 +54,22 @@ impl Default for DowngradeLeaderRegion {
 #[typetag::serde]
 impl State for DowngradeLeaderRegion {
     async fn next(&mut self, ctx: &mut Context) -> Result<(Box<dyn State>, Status)> {
-        let replay_timeout = ctx.persistent_ctx.replay_timeout;
         // Ensures the `leader_region_lease_deadline` must exist after recovering.
         ctx.volatile_ctx
             .set_leader_region_lease_deadline(Duration::from_secs(REGION_LEASE_SECS));
-        self.downgrade_region_with_retry(ctx).await;
+
+        match self.downgrade_region_with_retry(ctx).await {
+            Ok(_) => {
+                // Do nothing
+            }
+            Err(error::Error::ExceededDeadline { .. }) => {
+                // Rollbacks the metadata if procedure is timeout
+                return Ok((Box::new(UpdateMetadata::Rollback), Status::executing(false)));
+            }
+            Err(err) => {
+                warn!(err; "Occurs non-retryable error");
```

Comment:
No return when this non-retryable error occurred?

Reply:
Yes, if any non-retryable errors are incurred. The metasrv will wait for the target region lease to expire.

Reply:
Do we need to add an error log?

Reply:
Then the time wait below should be moved here.

Reply:
resolved in [927ff3c](https://github.com/GreptimeTeam/greptimedb/pull/4715/commits/927ff3c4457bde3919c0074dc815e775aaafbb94)